jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/RabitTracker.java

```diff
@@ -16,45 +17,40 @@
   // Maybe per tracker logger?
   private static final Log logger = LogFactory.getLog(RabitTracker.class);
   // tracker python file.
-  private static File tracker_py = null;
+  private static String tracker_py = null;
   // environment variable to be pased.
   private Map<String, String> envs = new HashMap<String, String>();
   // number of workers to be submitted.
   private int num_workers;
-  // child process
-  private  Process process = null;
-  // logger thread
-  private Thread logger_thread = null;
-
-  //load native library
-  static {
-    try {
-      initTrackerPy();
-    } catch (IOException ex) {
-      logger.error("load tracker library failed.");
-      logger.error(ex);
-    }
-  }
+  private AtomicReference<Process> trackerProcess = new AtomicReference<Process>();
 
   /**
    * Tracker logger that logs output from tracker.
    */
-  private class TrackerLogger implements Runnable {
+  private class TrackerProcessLogger implements Runnable {
     public void run() {
-      BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+
+      Log trackerProcessLogger = LogFactory.getLog(TrackerProcessLogger.class);
+      BufferedReader reader = new BufferedReader(new InputStreamReader(
+              trackerProcess.get().getErrorStream()));
       String line;
       try {
         while ((line = reader.readLine()) != null) {
-          logger.info(line);
+          trackerProcessLogger.info(line);
         }
       } catch (IOException ex) {
-        logger.error(ex.toString());
+        trackerProcessLogger.error(ex.toString());
       }
     }
   }
 
-  private static synchronized void initTrackerPy() throws IOException {
-    tracker_py = FileUtil.createTempFileFromResource("/tracker.py");
+  private void initTrackerPy() throws IOException {
+    try {
+      tracker_py = NativeLibLoader.createTempFileFromResource("/tracker.py");
```

Comment:
oh, I see, this is moved here. I am thinking only create tracker_py once, as tracker can be started multiple times